### ISSUE TITLE
Running doc example test suite through `cabal test` doesn't terminate

### ISSUE THREAD

**Issue by skykanin** [codedownio/sandwich#38]
When running the example test suite from your docs with `cabal test unit-tests` the tests don't terminate. Instead they get stuck on "RUNNING". However running `cabal run unit-tests -- --tui` works fine. I'm on GHC version 8.10.7 and `sandwich` version 0.1.0.8.

Code:
```hs
module Main where

import Test.Sandwich

basic :: TopSpec
basic = do
  describe "Arithmetic" $ do
    it "adds" $ do
      (2 + 2) `shouldBe` 4
      (2 + 3) `shouldBe` 5

    it "subtracts" $ do
      (3 - 2) `shouldBe` 0
      warn "This test might not be right..."

  describe "Strings" $
    it "concatenates" $
      ("abc" <> "def") `shouldBe` "abcdef"

main :: IO ()
main = runSandwichWithCommandLineArgs defaultOptions basic
```

CLI output:
```sh
Build profile: -w ghc-8.10.7 -O1
In order, the following will be built (use -v for more details):
 - Myopia-0.1.0.0 (test:unit-tests)  --enable-profiling (first run)
Preprocessing test suite 'unit-tests' for Myopia-0.1.0.0..
Building test suite 'unit-tests' for Myopia-0.1.0.0..
Running 1 test suites...
Test suite unit-tests: RUNNING...
```

**Comment by thomasjm** [comment]
I think this is probably the same as #35. Any setup that tries to start up the TUI from `cabal test`/`stack test` is going to run into this issue. 

I tried guide users toward the `run` versions in the docs, maybe you could point me to the part that led you astray?

**Comment by skykanin** [comment]
I'm not trying to run the tests with the TUI. I'm explicitly trying to run them without the TUI here when doing `cabal test`.

**Comment by thomasjm** [comment]
Okay, I think the check for whether there's a valid terminal attached wasn't working well. I just pushed 6d4db2044ca7cb588a6dd4cdef6b686f5773c08a, can you try it? It improves matters for me when using Stack.

**Comment by skykanin** [comment]
> Okay, I think the check for whether there's a valid terminal attached wasn't working well. I just pushed [6d4db20](https://github.com/codedownio/sandwich/commit/6d4db2044ca7cb588a6dd4cdef6b686f5773c08a), can you try it? It improves matters for me when using Stack.

Pointed cabal towards the commit and removed the constraint on `sandwich` in my `project.cabal` file:
```
source-repository-package
  type: git
  tag: 6d4db2044ca7cb588a6dd4cdef6b686f5773c08a
  location: git://github.com/codedownio/sandwich.git
  subdir: sandwich
```

However running `cabal test unit-tests` still doesn't terminate.

**Comment by thomasjm** [comment]
Okay, I think I figured it out. `cabal test` for whatever reason actually *does* connect the test process to a pty, but captures `stdout` nonetheless. So Sandwich thinks the TUI is running, but you don't see it.

Just pushed a solution in 4e46028e296db89638d354a392517fefa32fbaca. I decided the only reasonable thing is to default to the print formatter when using `runSandwichWithCommandLineArgs`, since I couldn't find any way to detect if we're running under `cabal test`.
